openshift-storage-libs/openshiftstoragelibs/baseclass.py

```diff
@@ -1206,3 +1212,180 @@ def validate_glusterfsd_memory_usage(self, size_limit):
+                    vol_name_prefix='scale', is_arbiter_vol=isarbiter,
+                    skip_cleanup=skip_cleanup)
+                storage_classes.append(sc_name)
+            else:
+                storage_classes.append(None)
+
+        sc_file, sc_arbiter, sc_block = storage_classes
+
+        # Create PVC's
+        pvcs = []
+        for sc, pvc_count, prefix in (
+                (sc_file, file_pvc_count, "auto-scale-pvc-file"),
+                (sc_arbiter, arbiter_pvc_count, "auto-scale-pvc-arbiter"),
+                (sc_block, block_pvc_count, "auto-scale-pvc-block")):
+            if pvc_count:
+                pvclist = self.create_pvcs_in_batch(
+                    sc, pvc_count,
+                    pvc_name_prefix="{}-{}".format(prefix, name_prefix),
+                    timeout=timeout, wait_step=wait_step,
+                    skip_cleanup=skip_cleanup)
+                pvcs.append(pvclist)
+            else:
+                pvcs.append(None)
+
+        pvcs_file, pvcs_arbiter, pvcs_block = pvcs
+
+        # Create app pods
+        for pvc_list, pod_count, prefix in (
+                (pvcs_file, file_pod_count, "file"),
+                (pvcs_arbiter, arbiter_pod_count, "arbiter"),
+                (pvcs_block, block_pod_count, "block")):
+            if pod_count:
+                self.create_app_pods_in_batch(
+                    pvc_list, pod_count, dc_name_prefix=(
+                        "auto-scale-dc-{}-{}".format(prefix, name_prefix)),
+                    timeout=timeout, wait_step=wait_step,
+                    skip_cleanup=skip_cleanup)
+
+    def verify_setup(self, prefix, timeout=300, wait_step=10):
+        # Verify all bricks are up
+        self.check_vol_status()
+
+        # Wait for heal
+        wait_to_heal_complete(timeout=timeout, wait_step=wait_step)
+
+        # Check mismatch of vols
+        match = (
+            'auto-scale-pvc-file-{}'.format(prefix),
+            'auto-scale-pvc-arbiter-{}'.format(prefix),
+            'auto-scale-pvc-block-{}'.format(prefix))
+
+        h_vol_list = heketi_volume_list(
+            self.heketi_client_node, self.heketi_server_url)
+        h_vol_list = [
+            vol.split('Name:')[1] for vol in h_vol_list.split('\n')
+            if any(m in vol for m in match)]
+
+        g_vol_list = get_volume_list('auto_get_gluster_endpoint')
+        g_vol_list = [
+            gvol for gvol in g_vol_list if any(m in gvol for m in match)]
+
+        pv_vols = oc_get_custom_resource(
+            self.ocp_master_node[0], 'pv', ':.spec.glusterfs.path')
+        pv_vols = [
+            vol[0] for vol in pv_vols
+            if vol[0] != '<none>' and any(m in vol[0] for m in match)]
+
+        self.assertTrue(
+            (h_vol_list.sort() == g_vol_list.sort() == pv_vols.sort()),
+            'Mismatch b/w heketi, gluster and pv volumes {}\n{}\n{}\n'.format(
+                h_vol_list, g_vol_list, pv_vols))
```

Comment:
Add comment for each block if needed.

Reply:
Upadated start of the block comment in detail

Reply:
Got it.

Reply:
Okay